template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/nifiregistry/NifiRegistryRoleConfigProvider.java

```diff
@@ -32,23 +41,33 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
         String cdhVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
                 "" : source.getBlueprintView().getProcessor().getStackVersion();
         if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_0_1)) {
-            RdsView nifiRegistryRdsView = getRdsView(source);
             switch (roleType) {
                 case NifiRegistryRoles.NIFI_REGISTRY_SERVER:
-                    return List.of(
-                            config(DATABASE_URL, nifiRegistryRdsView.getConnectionURL()),
-                            config(DRIVER_DIRECTORY, "/usr/share/java/"),
-                            config(DRIVER_CLASS, nifiRegistryRdsView.getConnectionDriver()),
-                            config(DATABASE_USER, nifiRegistryRdsView.getConnectionUserName()),
-                            config(DATABASE_PASSWORD, nifiRegistryRdsView.getConnectionPassword())
-                    );
+                    return generateDatabaseConfig(source);
                 default:
                     return List.of();
             }
         }
         return List.of();
     }
 
+    private List<ApiClusterTemplateConfig> generateDatabaseConfig(TemplatePreparationObject source) {
+        RdsView nifiRegistryRdsView = getRdsView(source);
+        List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+
+        configs.add(config(DATABASE_URL, nifiRegistryRdsView.getConnectionURL()));
+        configs.add(config(DRIVER_CLASS, nifiRegistryRdsView.getConnectionDriver()));
+        configs.add(config(DATABASE_USER, nifiRegistryRdsView.getConnectionUserName()));
+        configs.add(config(DATABASE_PASSWORD, nifiRegistryRdsView.getConnectionPassword()));
+
+        Optional<ClouderaManagerProduct> cfm = getCfmProduct(source);
+        if (cfm.isEmpty() || isVersionOlderThanLimited(cfm.get().getVersion(), CFM_VERSION_2_2_7_0)) {
```

Comment:
it is also running if there is no cfm. Is that right ?

Reply:
No, but currently this ConfigProvider is connected only to NifiRegistry Service Type (getServiceType method), which is only available for CFM product, so in theory this should never happens. I wonder before open the PR that is it worth that the Utility class return with Optional, because in theory Nifi and Nifi registry only available with CFM parcel, so if CFM is not available then the KnoxRoleConfigProvider and this should not be called at all (If yes then that would be a bug in the other code parts). 

What would be the best approach here? Change the code that if product is not found than throw runtime exception? I thought that if product is not availabe then falback to the old logic that is I wrote cfm.isEmpty also here. But I am not familiar with cloudbreak code convections. Do you guys prefer throw runtime exceptions or this project is much more like defensive programming style?